test/e2e/nixrun/run_suite_test.go

```diff
@@ -0,0 +1,173 @@
+package nixruntest
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/benchkram/bob/bob"
+	"github.com/benchkram/bob/bob/bobfile"
+	"github.com/benchkram/bob/pkg/file"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	dir     string
+	version string
+
+	stdout *os.File
+	stderr *os.File
+	pr     *os.File
+	pw     *os.File
+)
+
+type project struct {
+	name  string
+	gomod string
+	// .gox extension to avoid `main redeclared in this block`
+	main string
+}
+
+var _ = BeforeSuite(func() {
+	version = bob.Version
+	bob.Version = "1.0.0"
+
+	// Initialize mock bob files from local directory
+	bobFiles := []string{
+		"with_use_nix_false",
+		"init_with_bob_dependencies",
+		"init_with_task_dependencies",
+		"init_once_with_bob_dependencies",
+		"init_once_with_task_dependencies",
+		"binary_with_bob_dependencies",
+		"binary_with_task_dependencies",
+	}
+	nameToBobfile := make(map[string]*bobfile.Bobfile)
+	for _, name := range bobFiles {
+		abs, err := filepath.Abs("./" + name)
+		Expect(err).NotTo(HaveOccurred())
+		bf, err := bobfile.BobfileRead(abs)
+		Expect(err).NotTo(HaveOccurred())
+		nameToBobfile[strings.ReplaceAll(name, "/", "_")] = bf
+	}
+
+	testDir, err := ioutil.TempDir("", "bob-test-nix-run-*")
+	Expect(err).NotTo(HaveOccurred())
+
+	projects := []project{
+		{
+			name:  "server",
+			gomod: "./server/go.mod",
+			main:  "./server/main.go",
+		},
+		{
+			name:  "server-with-dep-inside",
+			gomod: "./server-with-dep-inside/go.mod",
+			main:  "./server-with-dep-inside/main.go",
+		},
+	}
+
+	// copy projects in the test dir
+	for _, p := range projects {
+		err = file.Copy(p.gomod, testDir+"/"+p.name+"_go.mod")
+		Expect(err).NotTo(HaveOccurred())
+
+		err = file.Copy(p.main, testDir+"/"+p.name+"_main.gox")
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	dir = testDir
+	err = os.Mkdir(dir+"/second_level", 0700)
+	Expect(err).NotTo(HaveOccurred())
+
+	err = os.Chdir(dir)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Save bob files in dir to have them available in tests
+	for name, bf := range nameToBobfile {
+		err = bf.BobfileSave(dir, name+".yaml")
+		Expect(err).NotTo(HaveOccurred())
+	}
+})
+
+var _ = AfterSuite(func() {
+	err := os.RemoveAll(dir)
+	Expect(err).NotTo(HaveOccurred())
+
+	for _, file := range tmpFiles {
+		err = os.Remove(file)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	bob.Version = version
+})
+
+func TestRun(t *testing.T) {
+	_, err := exec.LookPath("nix")
+	if err != nil {
+		// Allow skipping tests only locally.
+		// CI is always set to true on GitHub actions.
+		// https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables
+		if os.Getenv("CI") != "true" {
+			t.Skip("Test skipped because nix is not installed on your system")
+		}
+	}
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "nix run suite")
+}
+
+func startCapture() {
```

Comment:
Why do you name it differently in this test? Usually it was just named `capture()`. Please be consistent.

Reply:
I renamed it because here is a different behavior. When capture() is used then output() is used to get the captured output.

Here startCapture() just starts the capture, when the capturing is done in:

```go
			scanner := bufio.NewScanner(pr)
			Eventually(func() string {
				scanner.Scan()
				return scanner.Text()
			}, 5).Should(ContainSubstring("PHP 8.0.18"))
```

Also having close denotes the fact that it should that what was started should be closed(I can add a comment):

```go
			startCapture()
			defer closeCapture()
```

Maybe I can rename it to startScan or closeScan? I would keep them under a different name to not create confusion.

Reply:
Yes, please. Then rename it to `startScan` `stopScan`